From: John Fastabend <john.fastabend@gmail.com>
Date: Thu, 7 Dec 2017 09:57:59 -0800
Subject: net: skb_array: expose peek API
Patch-mainline: v4.16-rc1
Git-commit: 4a86a4cf68b4996a656d8c666617e8d839f8f839
References: bsc#1109837

This adds a peek routine to skb_array.h for use with qdisc.

Signed-off-by: John Fastabend <john.fastabend@gmail.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 include/linux/skb_array.h |    5 +++++
 1 file changed, 5 insertions(+)

--- a/include/linux/skb_array.h
+++ b/include/linux/skb_array.h
@@ -72,6 +72,11 @@ static inline bool __skb_array_empty(str
 	return !__ptr_ring_peek(&a->ring);
 }
 
+static inline struct sk_buff *__skb_array_peek(struct skb_array *a)
+{
+	return __ptr_ring_peek(&a->ring);
+}
+
 static inline bool skb_array_empty(struct skb_array *a)
 {
 	return ptr_ring_empty(&a->ring);
